File: fyp_namaste_events-main/lib/models/vendor_availability.dart
class VendorAvailability {
  final String id;
  final String vendorId;
  final DateTime date;
  final bool isAvailable;
  final String? reason;

  VendorAvailability({
    required this.id,
    required this.vendorId,
    required this.date,
    required this.isAvailable,
    this.reason,
  });

  factory VendorAvailability.fromJson(Map<String, dynamic> json) {
    return VendorAvailability(
      id: json['_id'],
      vendorId: json['vendorId'],
      date: DateTime.parse(json['date']),
      isAvailable: json['isAvailable'],
      reason: json['reason'],
    );
  }
}

File: fyp_namaste_events-main/backend/models/vendorAvailability.js
const mongoose = require("mongoose");

const vendorAvailabilitySchema = new mongoose.Schema({
  vendorId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'vendorModels',
    required: true
  },
  date: {
    type: Date,
    required: true
  },
  isAvailable: {
    type: Boolean,
    default: true
  },
  reason: {
    type: String
  },
  updatedAt: {
    type: Date,
    default: Date.now
  }
});

const vendorAvailabilityModel = mongoose.model("VendorAvailability", vendorAvailabilitySchema);
module.exports = { vendorAvailabilityModel };